src/Amadevus.RecordGenerator.Generators/CSharpKeyword.cs

```diff
@@ -0,0 +1,99 @@
+﻿namespace Amadevus.RecordGenerator.Generators
+{
+    static class CSharpKeyword
+    {
+        // The following method was derived from:
+        // Source https://github.com/dotnet/roslyn/blob/9d80dea7fe1b14043b9b2ac4d0b59ed26f508742/src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs#L1821-L1905
+        //
+        // Copyright (c) Microsoft. All Rights Reserved.
+        // Licensed under the Apache License, Version 2.0:
+        // https://www.apache.org/licenses/LICENSE-2.0
+
+        public static bool Is(this string name)
```

Comment:
I really don't like this approach, but I'm not sure there's any better way.

Reply:
So, I approached this problem already with https://github.com/amis92/RecordGenerator/commit/1172c747b1b1a5fd669b86299eca53ee21f5fc05 (using MS.CA.CSharp.Workspaces package). I'm considering keeping this PR open because I might be able to allow dependencies with CG.R soon, and then it'll be as nice as in the referenced commit.

Reply:
Ideally, Roslyn would have such a method. What your biggest concern? There have been no new keywords in C# for years; most if not all have been introduced as contextual keywords. It's not like this list needs maintaining a lot and it can be removed the day Roslyn has such an API. Meanwhile, happy to consider another approach you have to suggest.

Reply:
Let's leave it as it is for time being. I've created an issue for future enhancement in this area: #62